test/client_test.rb

```diff
@@ -53,12 +53,26 @@ def test_standard_values
     assert_equal(expected_quantiles, summary_metric.standard_values("value", "key")[:opts])
   end
 
+  def test_close_socket_on_error
+    logs = StringIO.new
+    logger = Logger.new(logs)
+    logger.level = :error
+
+    client = PrometheusExporter::Client.new(logger: logger, port: 321, once: true)
+    client.send("put a message in the queue")
+
+    assert_includes(
+      logs.string,
+      "Prometheus Exporter, failed to send message Connection refused - connect(2) for \"localhost\" port 321",
+    )
+  end
```

Comment:
This test does not directly test that the socket is closed but it does cause the internal `close_socket!` method to be called.

Reply:
Because it is an internal implementation detail and there is basically no way for us to assert for that in the test.